### PR TITLE
Add Haskell for Mac (Latest)

### PR DESCRIPTION
**Pull request by rodionovd**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked that the cask was not already refused in [closed issues].

[token reference]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/caskroom/homebrew-cask/pulls
[closed issues]: https://github.com/caskroom/homebrew-cask/issues?q=is%3Aissue+is%3Aclosed

### CODE REVIEW

File: Casks/haskell-for-mac.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+cask 'haskell-for-mac' do
+  version :latest
+  sha256 :no_check
+
+  # dl.devmate.com/com.haskellformac.Haskell.basic was verified as official when first introduced to the cask
+  url 'https://dl.devmate.com/com.haskellformac.Haskell.basic/Haskell%E2%80%94FunctionalProgrammingLab.dmg'
+  name 'Haskell for Mac'
+  homepage 'http://haskellformac.com'
+  license :commercial
```
Comment (miccal) [inline Casks/haskell-for-mac.rb]: Please remove the `;licence` stanza; it has been deprecated.

Comment (rodionovd) [inline Casks/haskell-for-mac.rb]: Oh thanks! I missed that somehow

File: Casks/haskell-for-mac.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+cask 'haskell-for-mac' do
+  version :latest
+  sha256 :no_check
+
+  # dl.devmate.com/com.haskellformac.Haskell.basic was verified as official when first introduced to the cask
+  url 'https://dl.devmate.com/com.haskellformac.Haskell.basic/Haskell%E2%80%94FunctionalProgrammingLab.dmg'
```
Comment (miccal) [inline Casks/haskell-for-mac.rb]: Please change this to `url 'https://dl.devmate.com/com.haskellformac.Haskell.basic/Haskell—FunctionalProgrammingLab.dmg'`.

Comment (rodionovd) [inline Casks/haskell-for-mac.rb]: The thing is that it's not a dash but *a long dash* (aka en dash) so it has to be encoded I believe

Comment (miccal) [inline Casks/haskell-for-mac.rb]: You are correct.

**Comment by vitorgalvao** [comment]
This is a MAS-only app. As far as I know, you cannot buy it outside it. We do not accept those apps.

**Comment by rodionovd** [comment]
@vitorgalvao you're not quite right: there's [an option](http://store.haskellformac.com) to buy this app outside of MAS via Fastspring

File: Casks/haskell-for-mac.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  name 'Haskell for Mac'
+  homepage 'http://haskellformac.com'
+
+  auto_updates true
```
Comment (vitorgalvao) [inline Casks/haskell-for-mac.rb]: If an app auto-updates it has an [`appcast`](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/appcast.md) somewhere. Chan you figure out what it is?

Comment (rodionovd) [inline Casks/haskell-for-mac.rb]: Sure. I'll ping you when it's done

Comment (vitorgalvao) [inline Casks/haskell-for-mac.rb]: > I just thought that it doesn't make sense to have an appcast with `:latest`

If you find the appcast, change `version` and `sha256` to actual values. We’re working on tools to take appcasts and detect changes early on, so in those cases they are versioned.

**Review by vitorgalvao — APPROVED** [review]

**Comment by rodionovd** [comment]
@vitorgalvao thanks! Sorry it took me a while to find an appcast, but this app doesn't use Sparkle directly so it required a bit of reverse-engineering :)

**Comment by vitorgalvao** [comment]
> Sorry it took me a while to find an appcast: this app doesn't use Sparkle directly so it required a bit of reverse-engineering :)

No apologies needed. Good work finding it!

**Review by vitorgalvao — APPROVED** [review]

**Pull request merged**